src/scenes/Engagement/InternshipEngagement/Create/CreateInternshipEngagement.tsx

```diff
@@ -0,0 +1,56 @@
+import { useSnackbar } from 'notistack';
+import React from 'react';
+import { Except } from 'type-fest';
+import { ButtonLink } from '../../../../components/Routing';
+import { ProjectOverviewDocument as ProjectOverview } from '../../../Projects/Overview/ProjectOverview.generated';
+import { useCreateInternshipEngagementMutation } from './CreateInternshipEngagement.generated';
+import {
+  CreateInternshipEngagementForm,
+  CreateInternshipEngagementFormProps as Props,
+} from './CreateInternshipEngagementForm';
+
+type CreateInternshipEngagementProps = Except<Props, 'onSubmit'> & {
+  projectId: string;
+};
+
+export const CreateInternshipEngagement = ({
+  projectId,
+  ...props
+}: CreateInternshipEngagementProps) => {
+  const [createEngagement] = useCreateInternshipEngagementMutation();
+  const { enqueueSnackbar } = useSnackbar();
+  const submit: Props['onSubmit'] = async (input) => {
+    const createInternshipEngagementInput = {
+      engagement: { projectId, internId: input.users.id },
+    };
+    const res = await createEngagement({
+      variables: { input: createInternshipEngagementInput },
+      refetchQueries: [
+        {
+          query: ProjectOverview,
+          variables: { input: projectId },
+        },
+      ],
+    });
+
+    const engagement = res.data!.createInternshipEngagement.engagement;
+
+    enqueueSnackbar(
+      `Created internship for:
```

Comment:
```suggestion
      `Created internship engagement for:
```
or 
```suggestion
      `Created engagement for:
```

We don't have an "Internship"s. We have Projects and Engagements. If any label would get Internship it would be the at the Project level I believe, not the engagement level.